src/container/srv_target.c

```diff
@@ -1630,59 +1630,55 @@ ds_cont_local_open(uuid_t pool_uuid, uuid_t cont_hdl_uuid, uuid_t cont_uuid,
 		 */
 		D_ASSERT(hdl->sch_cont != NULL);
 		D_ASSERT(hdl->sch_cont->sc_pool != NULL);
-
 		hdl->sch_cont->sc_open++;
```

Comment:
Suggestion: simply moving ABT_mutex_lock() before sc_open++ would make code simpler, and you don't have to change the assertions in DTX code.

Reply:
> Suggestion: simply moving ABT_mutex_lock() before sc_open++ would make code simpler, and you don't have to change the assertions in DTX code.

Yes, I would suggest for the same. On the other hand, what will happen if another open arrives with the same uuid during former one yield? The latter one will return directly since it can find the "hdl" in cache but without considering whether the container is really opened or in-opening. That seems incorrect.

Reply:
@wangdi1 Could you update the PR to address this comment? Thanks!